ironfish/src/rpc/routes/wallet/createTransaction.ts

```diff
@@ -0,0 +1,225 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Asset } from '@ironfish/rust-nodejs'
+import { BufferMap } from 'buffer-map'
+import * as yup from 'yup'
+import { BurnDescription } from '../../../primitives/burnDescription'
+import { MintDescription } from '../../../primitives/mintDescription'
+import { RawTransactionSerde } from '../../../primitives/rawTransaction'
+import { CurrencyUtils } from '../../../utils'
+import { NotEnoughFundsError } from '../../../wallet/errors'
+import { ERROR_CODES, ValidationError } from '../../adapters/errors'
+import { ApiNamespace, router } from '../router'
+
+export type CreateTransactionRequest = {
+  fromAccountName: string
+  receives: {
+    publicAddress: string
+    amount: string
+    memo: string
+    assetId?: string
+  }[]
+  mints?: {
+    privateKey: string
+    name: string
+    metadata: string
+    value: string
+  }[]
+  burns?: {
+    assetId: string
+    value: string
+  }[]
+  fee?: string
+  feeRate?: string
+  expiration?: number
+  expirationDelta?: number
+}
+
+export type CreateTransactionResponse = {
+  transaction: string
+}
+
+export const CreateTransactionRequestSchema: yup.ObjectSchema<CreateTransactionRequest> = yup
+  .object({
+    fromAccountName: yup.string().defined(),
+    receives: yup
+      .array(
+        yup
+          .object({
+            publicAddress: yup.string().defined(),
+            amount: yup.string().defined(),
+            memo: yup.string().defined(),
+            assetId: yup.string().optional(),
+          })
+          .defined(),
+      )
+      .defined(),
+    mints: yup
+      .array(
+        yup
+          .object({
+            privateKey: yup.string().defined(),
+            name: yup.string().defined(),
+            metadata: yup.string().defined(),
+            value: yup.string().defined(),
+          })
+          .defined(),
+      )
+      .optional(),
+    burns: yup
+      .array(
+        yup
+          .object({
+            assetId: yup.string().defined(),
+            value: yup.string().defined(),
+          })
+          .defined(),
+      )
+      .optional(),
+    fee: yup.string().optional(),
+    feePriorityLevel: yup.string().optional(),
+    expiration: yup.number().optional(),
+    expirationDelta: yup.number().optional(),
+  })
+  .defined()
+
+export const CreateTransactionResponseSchema: yup.ObjectSchema<CreateTransactionResponse> = yup
+  .object({
+    transaction: yup.string().defined(),
+  })
+  .defined()
+
+router.register<typeof CreateTransactionRequestSchema, CreateTransactionResponse>(
+  `${ApiNamespace.wallet}/createTransaction`,
+  CreateTransactionRequestSchema,
+  async (request, node): Promise<void> => {
+    const transaction = request.data
```

Comment:
this isnt a transaction, lets delete this variable and just use request.data, or just call this data.